native/src/main/java/io/ballerina/wso2/apim/catalog/ServiceCatalog.java

```diff
@@ -0,0 +1,265 @@
+        HttpServiceConfig httpServiceConfig = updateHostAndPortAndBasePath(listenerDetails,
+                attachPointDetails, httpAnnotation);
+        updateServiceNameAndUrl(artifactValues, httpServiceConfig);
+        updateAnnotationsArtifactValues(artifactValues, annotationDetails, httpAnnotation, env, httpServiceConfig);
+        updateListenerConfigurations(artifactValues, listenerDetails);
+        updateMd5(artifactValues);
+    }
+
+    private static void updateMd5(BMap<BString, Object> artifactValues) {
+        String name = StringUtils.getStringValue(artifactValues.get(StringUtils.fromString(NAME)));
+        String version = StringUtils.getStringValue(artifactValues.get(StringUtils.fromString(VERSION)));
+        String definitionType = StringUtils.getStringValue(artifactValues.get(StringUtils.fromString(DEFINITION_TYPE)));
+        String openapiDef = StringUtils.getStringValue(artifactValues.get(StringUtils
+                .fromString(DEFINITION_FILE_CONTENT)));
+
+        String string = new StringBuilder(name).append(version).append(definitionType).append(openapiDef).toString();
+        artifactValues.put(StringUtils.fromString(MD5), StringUtils.fromString(createMd5Hash(string)));
+    }
+
+
+    private static void updateServiceNameAndUrl(BMap<BString, Object> artifactValues,
+                                                HttpServiceConfig httpServiceConfig) {
+        updateServiceName(artifactValues, httpServiceConfig);
+        updateServiceUrl(artifactValues, httpServiceConfig);
+    }
+
+    private static void updateListenerConfigurations(BMap<BString, Object> artifactValues, Object listenerDetails) {
+        artifactValues.put(StringUtils.fromString(MUTUAL_SSL_ENABLED), getMutualSSLDetails(listenerDetails));
+    }
+
+
+    private static boolean getMutualSSLDetails(Object listenerDetails) {
+        ArrayList<BObject> listenerArray = (ArrayList<BObject>) listenerDetails;
+        if (listenerArray.size() > 0) {
+            BObject listener = listenerArray.get(0);
+            if (listener != null) {
+                Object configObject = listener.getNativeData(CONFIG);
+                if (configObject instanceof HashMap) {
+                    HashMap<BString, BString> config = (HashMap<BString, BString>) configObject;
+                    if (config.containsKey(StringUtils.fromString(SECURE_SOCKET))) {
+                        Object secureSocketObject = config.get(StringUtils.fromString(SECURE_SOCKET));
+                        if (secureSocketObject instanceof HashMap) {
+                            HashMap<BString, BString> secureSocket =
+                                    (HashMap<BString, BString>) secureSocketObject;
+                            if (secureSocket.containsKey(StringUtils.fromString(MUTUAL_SSL))) {
+                                return true;
+                            }
+                        }
+                    }
+                }
```

Comment:
I think you can use pattern matching `instanceOf` to simplify this,
```suggestion
                if (configObject instanceof HashMap<BString, BString> config && config.containsKey(StringUtils.fromString(SECURE_SOCKET))) {
                    Object secureSocketObject = config.get(StringUtils.fromString(SECURE_SOCKET));
                    if (secureSocketObject instanceof HashMap<BString, BString> secureSocket && secureSocket.containsKey(StringUtils.fromString(MUTUAL_SSL))) {
                        return true;
                    }
                }

```